src/chromap.cc

```diff
@@ -150,8 +168,19 @@ void Chromap::TrimAdapterForPairedEndRead(uint32_t pair_index,
         // Trim adapters and TODO: fix sequencing errors
         int overlap_length =
             read2_length - seed_start_position + si * seed_length;
-        read_batch1.TrimSequenceAt(pair_index, overlap_length);
-        read_batch2.TrimSequenceAt(pair_index, overlap_length);
+        // The case that read1 is contained in read2
+        int read2_offset = 0; 
+        if (overlap_length > (int)read1_length) {
+          read2_offset = overlap_length - read1_length;
+          overlap_length = read1_length;
+        }
+        if (read1_length <= read2_length) {
```

Comment:
We make read1 always the shorter one. Then why would we have two cases here (r1<=r2 and r1>r2)?